lib/pbench/server/auth/auth.py

```diff
@@ -143,14 +143,15 @@ def verify_auth_oidc(auth_token: str) -> Optional[User]:
     """
     try:
         token_payload = oidc_client.token_introspect(token=auth_token)
-    except OpenIDTokenInvalid:
-        # The token is not a valid access token, fall through.
-        pass
     except Exception:
-        current_app.logger.exception(
-            "Unexpected exception occurred while verifying the auth token {}",
-            auth_token,
-        )
+        try:
+            return verify_auth_api_key(auth_token)
+        except Exception:
+            current_app.logger.exception(
+                "Unexpected exception occurred while verifying the API key {}",
+                auth_token,
+            )
+        raise
```

Comment:
Previously (at old line 150) we would log any "unexpected" errors received when attempting to validate the OIDC access token; now when that happens, we simply re-raise the exception.  (That is, if the API Key validation (also) fails unexpectedly, we log it, but if the OIDC access token fails, we don't...at least in this code....)  Does the caller log it?  (Were we previously logging it _twice_??)

Is this expected/intended/goodness?

Reply:
Yeah; the caller is `verify_auth`, which logs unexpected exceptions, so we were previously logging it twice. I'm logging the "inner" API key exception which would otherwise be hidden, but letting the caller handle the "outer" OIDC error before (otherwise) silently ignoring that some fool attempted authentication in the first place.